pkg/mock/mockcluster/mockcluster.go

```diff
@@ -216,6 +216,19 @@ func (mc *Cluster) AddRegionStore(storeID uint64, regionCount int) {
 	mc.PutStore(store)
 }
 
+// AddRegionStoreWithLeader adds store with specified count of region and leader.
+func (mc *Cluster) AddRegionStoreWithLeader(storeID uint64, regionCount int, leaderCounts ...int) {
```

Comment:
Can we use `AddRegionStore` and `UpdateLeaderCount` to do it or some existing methods?

Reply:
the two method only can update region num rather than adding region